lib/api/CDataFrameAnalyzer.cc

```diff
@@ -254,28 +255,30 @@ void CDataFrameAnalyzer::addRowToDataFrame(const TStrVec& fieldValues) {
     });
 }
 
-void CDataFrameAnalyzer::reportProgress(const CDataFrameAnalysisRunner& analysis,
-                                        core::CRapidJsonConcurrentLineWriter& writer) const {
-    double progress{0.0};
+void CDataFrameAnalyzer::monitorProgress(const CDataFrameAnalysisRunner& analysis,
+                                         core::CRapidJsonConcurrentLineWriter& writer) const {
+    // Progress as percentage
+    int progress{0};
     while (analysis.finished() == false) {
         std::this_thread::sleep_for(std::chrono::milliseconds(100));
-        double latestProgress{analysis.progress()};
+        int latestProgress{static_cast<int>(floor(100.0 * analysis.progress()))};
         if (latestProgress > progress) {
             progress = latestProgress;
             this->writeProgress(progress, writer);
         }
     }
-    if (progress < 1.0) {
-        this->writeProgress(1.0, writer);
+    if (progress < 100) {
```

Comment:
nit: this if block isn't strictly needed. The managing of setting to 1.0 is handled in the exit condition of task runner, see [here](https://github.com/elastic/ml-cpp/blob/master/lib/api/CDataFrameAnalysisRunner.cc#L150) and its call [here](https://github.com/elastic/ml-cpp/blob/master/lib/api/CDataFrameAnalysisRunner.cc#L103).